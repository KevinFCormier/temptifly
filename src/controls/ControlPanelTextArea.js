'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { FormGroup, TextArea, Popover } from '@patternfly/react-core'
import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon'

class ControlPanelTextArea extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    controlId: PropTypes.string,
    handleChange: PropTypes.func,
    i18n: PropTypes.func
  };

  constructor(props) {
    super(props)
    this.state = {}
  }

  setControlRef = (control, ref) => {
    control.ref = ref
  };

  render() {
    const { controlId, i18n, control } = this.props
    const {
      name,
      type,
      active: value,
      exception,
      tooltip,
      validation = {},
      disabled
    } = control

    // if placeholder missing, create one
    let { placeholder } = control
    if (!placeholder) {
      placeholder = i18n(
        'creation.ocp.cluster.enter.value',
        [name ? name.toLowerCase() : '']
      )
    }

    const validated = exception ? 'error' : undefined
    return (
      <React.Fragment>
        <div
          className="creation-view-controls-textbox"
          style={{ display: '' }}
          ref={this.setControlRef.bind(this, control)}
        >
          <FormGroup
            id={`${controlId}-label`}
            label={name}
            isRequired={validation.required}
            fieldId={controlId}
            helperTextInvalid={exception}
            validated={validated}
            labelIcon={
              /* istanbul ignore next */
              tooltip ? (
                <Popover
                  id={`${controlId}-label-help-popover`}
                  bodyContent={tooltip}
                >
                  <button
                    id={`${controlId}-label-help-button`}
                    aria-label="More info"
                    onClick={(e) => e.preventDefault()}
                    className="pf-c-form__group-label-help"
                  >
                    <HelpIcon noVerticalAlign />
                  </button>
                </Popover>
              ) : (
                <React.Fragment />
              )
            }
          >
            <TextArea
              id={controlId}
              disabled={disabled}
              type={type}
              placeholder={placeholder}
              validated={validated}
              value={value || ''}
              onChange={this.handleChange.bind(this, control)}
            />
          </FormGroup>
        </div>
      </React.Fragment>
    )
  }

  handleChange(id, evt) {
    const { control, handleChange } = this.props
    control.active = evt
    handleChange(evt)
  }
}

export default ControlPanelTextArea
