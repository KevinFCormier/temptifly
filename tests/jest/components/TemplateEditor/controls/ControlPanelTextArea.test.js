'use strict'

import React from 'react'
import ControlPanelTextArea from '../../../../../src/controls/ControlPanelTextArea'
import renderer from 'react-test-renderer'

import { control } from '../../TestingData'

describe('ControlPanelTextArea component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <ControlPanelTextArea
        key={'key'}
        control={control}
        controlId={'controlId'}
        handleChange={fn}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
