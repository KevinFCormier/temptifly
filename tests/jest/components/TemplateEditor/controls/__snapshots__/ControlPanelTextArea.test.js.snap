// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanelTextArea component renders as expected 1`] = `
<div
  className="creation-view-controls-textarea"
>
  <div
    className="creation-view-controls-textarea-title"
  >
    creation.app.name
    <div
      className="bx--tooltip__label"
    >
      
      <div
        aria-expanded={false}
        aria-haspopup="true"
        aria-label="tooltip"
        className="bx--tooltip__trigger"
        id=""
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="button"
        tabIndex={0}
        title=""
      >
        <svg
          alt="tooltip"
          aria-label="tooltip"
          className={undefined}
          fill={undefined}
          fillRule="evenodd"
          height="16"
          name={undefined}
          role="img"
          style={undefined}
          viewBox="0 0 16 16"
          width="16"
        >
          <title>
            
          </title>
          <path
            d="M8,14.5 C11.5898509,14.5 14.5,11.5898509 14.5,8 C14.5,4.41014913 11.5898509,1.5 8,1.5 C4.41014913,1.5 1.5,4.41014913 1.5,8 C1.5,11.5898509 4.41014913,14.5 8,14.5 Z M8,16 C3.581722,16 0,12.418278 0,8 C0,3.581722 3.581722,0 8,0 C12.418278,0 16,3.581722 16,8 C16,12.418278 12.418278,16 8,16 Z"
          />
          <polygon
            points="9 13 7 13 7 7 9 7"
          />
          <circle
            cx="8"
            cy="4"
            r="1"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="bx--form-item"
  >
    <textarea
      aria-describedby={null}
      aria-invalid={null}
      autoComplete="new-password"
      className="bx--text-area"
      cols={50}
      data-invalid={null}
      disabled={false}
      id="controlId"
      onChange={[Function]}
      onClick={[Function]}
      placeholder=""
      rows={4}
      spellCheck={false}
      value="test"
    />
  </div>
</div>
`;
